README.md

```diff
@@ -69,4 +69,42 @@ end)
 
 ## Step 4: Test the Implementation
 
-At this point, the SDK has been installed, initialized, authenticated and bound to a UI button. The last step is to test the implementation to ensure it has been integrated properly.
\ No newline at end of file
+At this point, the SDK has been installed, initialized, authenticated and bound to a UI button. The last step is to test the implementation to ensure it has been integrated properly.
+
+# Styngr Backend
+
+The Styngr backend consists of a set of APIs enabling clients to build radio-like user experiences that are in line with the DMCA rulesets.
+
+Styngr takes care of the legal and licensing aspects of dealing with the major labels which enables the game developers to focus on implementing the best experiences for their users.
+
+The geo-availability of audio assets is dictated by the label owners, who also reserve full rights to add support for new territories or remove support for the existing ones.
+
+## Geo-blocking
+
+Depending on the client's geolocation, access to certain content provided by Styngr may be restricted. To achieve this, Styngr stores clients' geolocation in the JWT when they are successfully authenticated and uses this information every time a request comes in.
+
+Based on the client's technical implementation and architecture, Styngr offers two following solutions for geo-blocking:
+
+1. Styngr APIs approximate the client's geolocation by looking up the source IP address of every HTTP request. The user's location is determined by searching for the public IP address in Styngr's database which is periodically synchronized with data from MaxMind.
+
+2. In case when the client application or the SDK integrates with Styngr APIs via a proxy server, the client's geolocation can no longer reliably be determined by looking up the source IP address in the HTTP request. Therefore, the client is advised to use the `V2` version of the token endpoint (`POST /api/v2/sdk/tokens`). This endpoint expects the `countryCode` in ISO 3166, alpha-2 format to be passed in the request body.
+
+### How It Works
+
+1.	Client authenticates with the Styngr backend by calling one of the available token endpoint versions. 
+2.	Styngr API extracts the client's public IP address from the HTTP request when `V1` token endpoint is called or uses the `countryCode` value from the HTTP request body when the `V2` token endpoint is called.
+3.	If the client authenticated via `V1` token endpoint, Styngr API uses the previously extracted public IP address to approximate the client’s geolocation by looking up the MaxMind database from which the country information is retrieved.
+4.	Styngr API cross-references the country information with internal database that holds tracks metadata to determine whether there are any applicable geographical restrictions for the requested audio asset.
```

Comment:
I understand this (and following points) as user's already knows which track they want to request from our system, when in reality, user will receive a track that is randomly selected. 

Now when I think about it, isn't Roblox going to use their own randomization algorithm and wouldn't they load track from their cloud storage?

Reply:
I didn't inspect the SDK code to such extent (to check whether they do some kind of randomization as well), but the simple fact that they always call the `Next` API endpoint to retrieve the next track probably means they rely on Styngr's randomization algorithm.

Reply:
They still do load actual audio files from the Roblox server. However, I think the randomization algorithm is completely on Styngr side.

We simply return the next `rblxassetid://...` for them to use.